lib/parsers/tiff_parser.rb

```diff
@@ -22,6 +22,9 @@ def call(io)
     exif_data = exif_from_tiff_io(io)
     return unless exif_data
 
+    # Skip the parsing if the image is Sony arw format
+    return if arw?(exif_data)
+
     w = exif_data.width || exif_data.pixel_x_dimension
     h = exif_data.height || exif_data.pixel_y_dimension
```

Comment:
I wonder if we still need the fallbacks if we skip ARWs. In particular because you also removed the test for this 😅  See #146

Reply:
Do you think it would be better to add ARW parser and redirect the parsing to it, to handle this? I am a bit confused because we do a similar thing(ignore parsing) for tiff files with cr2 format.

Reply:
@martijnvermaat I didn't get what you mean about the fallbacks..

FormatParser runs [all the parsers](https://github.com/WeTransfer/format_parser/blob/master/lib/format_parser.rb#L170) no matter what is the result of each one.

So, I think there is no way to "disable" the fallbacks.

Maybe the fallback that you mentioned is not the same that I'm thinking of. 🤔

Reply:
@fabioperrella With fallbacks I meant using `pixel_x_dimension` if `width` is not available (same for `y`). I think this is only needed for ARW files, which we are now skipping. Although on closer inspection I see that the original code (before #146 ) used `image_width` so I'm a bit confused.

@nitika080289 I'm not sure what the best way would be to add actual support for ARW. As a separate parser, or to add it to the TIFF parser. But that's something for another PR I'd say, first good step is to stop parsing ARW because we can't really handle it well (for now).

Reply:
Checked the backend code and looks like these changes should work because now the format parser result would be nil. So the error in #146 would not be raised.